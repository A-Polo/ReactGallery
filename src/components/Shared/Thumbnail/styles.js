export default {
  Album: {
    margin: [0, 10],
  },
};
