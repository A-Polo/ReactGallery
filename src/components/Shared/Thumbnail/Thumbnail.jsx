import React from 'react';
import PropTypes from 'prop-types';
import injectSheet from 'react-jss';

import styles from './styles';

const Thumbnail = ({
  onClick, onKeyPress, children, classes,
}) => (
  <div className={classes.Album} onClick={onClick} onKeyPress={onKeyPress} role="button" tabIndex="0">
    { children }
  </div>
);

Thumbnail.propTypes = {
  onClick: PropTypes.func.isRequired,
  onKeyPress: PropTypes.func.isRequired,
  children: PropTypes.instanceOf(Object).isRequired,
  classes: PropTypes.instanceOf(Object).isRequired,
};

export default injectSheet(styles)(Thumbnail);
