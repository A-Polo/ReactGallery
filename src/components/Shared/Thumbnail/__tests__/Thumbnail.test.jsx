import React from 'react';
import { shallow } from 'enzyme';
import Thumbnail from '../Thumbnail';

describe('components/Button', () => {
  const testProps = {
    onClick: () => {},
    onKeyPress: () => {},
  };

  describe('props', () => {
    describe('.onClick', () => {
      it('is being executed when user clicks on the Thumbnail', () => {
        jest.spyOn(testProps, 'onClick');

        const wrapper = shallow(<Thumbnail {...testProps}><h2>Title</h2><img src="http://test.src" alt="testAlt" /></Thumbnail>);
        wrapper.simulate('click');

        expect(testProps.onClick).toHaveBeenCalled();
      });
    });

    describe('.onKeyPress', () => {
      it('should be executed when user press any key', () => {
        jest.spyOn(testProps, 'onKeyPress');

        const wrapper = shallow(<Thumbnail {...testProps}><h2>Title</h2><img src="http://test.src" alt="testAlt" /></Thumbnail>);
        wrapper.simulate('keypress');

        expect(testProps.onKeyPress).toHaveBeenCalled();
      });
    });
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<Thumbnail {...testProps}><h2>Title</h2><img src="http://test.src" alt="testAlt" /></Thumbnail>);
    expect(wrapper).toMatchSnapshot();
  });
});
