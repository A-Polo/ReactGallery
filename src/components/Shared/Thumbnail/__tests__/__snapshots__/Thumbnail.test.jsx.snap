// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Button matches snapshot 1`] = `
<Thumbnail
  classes={
    Object {
      "Album": "Thumbnail-Album-0-1-1",
    }
  }
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  onKeyPress={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <h2>
    Title
  </h2>
  <img
    alt="testAlt"
    src="http://test.src"
  />
</Thumbnail>
`;
